services/streammanager/kafka/client/producer.go

```diff
@@ -109,17 +109,9 @@ func (p *Producer) Close(ctx context.Context) error {
 func (p *Producer) Publish(ctx context.Context, msgs ...Message) error {
 	messages := make([]kafka.Message, len(msgs))
 	for i := range msgs {
-		var headers []kafka.Header
-		if l := len(msgs[i].Headers); l > 0 {
-			headers = make([]kafka.Header, l)
-			for k := range msgs[i].Headers {
-				headers[k] = kafka.Header{
-					Key:   msgs[i].Headers[k].Key,
-					Value: msgs[i].Headers[k].Value,
-				}
-			}
-		}
+		headers := processHeaderForEachMessage(msgs[i])
 		messages[i] = kafka.Message{
+			Topic:   msgs[i].Topic,
```

Comment:
`client.Producer` should know if it has been initialized with a topic or not and only include the message's topic in the latter case.
Additionally, this function should return an error if the producer is not initialized with a topic and no topic is included in any of the messages.

Reply:
We plan to go with the approach where we always initialize the producer without client and we intend to always send each message with topic information. That proper assignment of topic to each message would be handled at transformer

Reply:
Given that we decided to go down this road, can you please make it mandatory and raise an error here if the topic is empty?

```go
	for i := range msgs {
		if msgs[i].Topic == "" {
			return fmt.Errorf("no topic provided for message %d", i)
		}
```

I think we should also have a test for this (don't specify topic and check that an error is returned).